src/robot/running/context.py

```diff
@@ -124,6 +126,16 @@ def in_teardown(self):
     def variables(self):
         return self.namespace.variables
 
+    @property
+    def continue_on_failure(self):
+        kw_or_test = self.user_keywords[-1] if self.user_keywords else self.test
+        if not kw_or_test:
+            return False
+        if 'robot:continue-on-failure' in kw_or_test.tags:
+            return True
+        return self.test and 'robot:continue-on-failure-recursive' in self.test.tags \
+            or any([kw for kw in self.user_keywords if 'robot:continue-on-failure-recursive' in kw.tags])
```

Comment:
This could be simplified a bit. Something like this (untested):
```python
parents = ([self.test] if self.test or []) + self.user_keywords
if not parents:
    return False
if 'robot:continue-on-failure' in parents[-1].tags:
    return True
return any('robot:continue-on-failure-recursive' in p.tags for p in parents)
```

Reply:
I didn't like `parents` as a var name as the list includes the current context too, but I am fine if you are fine.. thanks for the suggestion, much nicer!